### ISSUE TITLE
AppCenter error when building Android

### ISSUE THREAD

**Issue by jkoutavas** [jkoutavas/ReactNative-AppCenter#1]
When building the most basic React Native app (`npx react-native init AwesomeProject`) we get this error when building on AppCenter:

```
##[section]Starting: Gradle Task
==============================================================================
Task         : Gradle
Description  : Build using a Gradle wrapper script
Version      : 1.128.0
Author       : Microsoft Corporation
Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613720)
==============================================================================
SYSTEMVSSCONNECTION exists true
[command]/Users/runner/runners/2.160.1/work/1/s/AwesomeProject/android/gradlew -p AwesomeProject/android clean :app:assembleRelease
Starting a Gradle Daemon, 1 incompatible and 1 stopped Daemons could not be reused, use --status for details
Gradle 5.5

FAILURE: Build failed with an exception.

* Where:
Settings file '/Users/runner/runners/2.160.1/work/1/s/AwesomeProject/android/settings.gradle' line: 2

* What went wrong:
A problem occurred evaluating settings 'AwesomeProject'.
> Could not read script '/Users/runner/runners/2.160.1/work/1/s/AwesomeProject/node_modules/@react-native-community/cli-platform-android/native_modules.gradle' as it does not exist.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 7s
Error: /Users/runner/runners/2.160.1/work/1/s/AwesomeProject/android/gradlew failed with return code: 1
    at ChildProcess.<anonymous> (/Users/runner/runners/2.160.1/work/_tasks/Gradle_8d8eebd8-2b94-4c97-85af-839254cc6da4/1.128.0/node_modules/vsts-task-lib/toolrunner.js:569:30)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:886:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
##[error]Error: /Users/runner/runners/2.160.1/work/1/s/AwesomeProject/android/gradlew failed with return code: 1

```

**Comment by jkoutavas** [comment]
Adding `cli-platform-android` to the project  (`yarn add @react-native-community/cli-platform-android`) did not fix the issue.  See branch `cli-platform-android`

**Comment by ddolheguy** [comment]
I can confirm App Center is completely broken with the latest Android RN.

**Comment by geekazoid-at** [comment]
Following! Also having issues related with the CLI toolset with 0.61.5. I tried the proposed fix with no success.

**Comment by jkoutavas** [comment]
@geekazoid-at, @ddolheguy 

I got a response from AppCenter support yesterday:

"Hi There, I had discussed with the build team, there is an issue with  react-native-community/cli-platform-android.

The problem which seems to be that the directory to execute from is not provided, which is a change coming from this commit in the React Native Android CLI tooling: https://github.com/react-native-community/cli/commit/1799e4440123821db1085517695c51ad7c2dacb8

Based on the discussion here: https://github.com/react-native-community/cli/pull/824 this change still seems to be causing discussions and the maintainers are considering to potentially undo it in its entirety: https://github.com/react-native-community/cli/pull/852

This would be part of a new 3.0.x release of the @react-native-community/cli-platform-android package but is not released at this point. So actually, a potential fix might be to go back to a release prior to that, i.e. installing @react-native-community/cli-platform-android@3.0.0-alpha.6 - but I can't guarantee it'll work - it will also have negative impact on building your app on Windows since the CLI has a bug there until alpha.7.

In the meantime given all the different complex issues with this command - while this will hopefully be fixed soon, - when you're encountering this case, React Native 0.60.x might be the safer bet to build."

**Comment by ddolheguy** [comment]
@jkoutavas 

Thanks for this update, difficult one.  We're manually building this for the moment whilst we consider a solution, but going back to 0.60.x might be the solution for the time being.

**Comment by jkoutavas** [comment]
Have a look at the comment that was added here today:

https://github.com/microsoft/appcenter/issues/1285

**Comment by geekazoid-at** [comment]
> Have a look at the comment that was added here today:
> 
> [microsoft/appcenter#1285](https://github.com/microsoft/appcenter/issues/1285)

I revised my statement here: https://github.com/microsoft/appcenter/issues/1285#issuecomment-569059608
Seemed to fix the build but caused other issues which I did not manage to work around. I suppose because cli versions 2 are not compatible with RN61 (https://github.com/react-native-community/cli).

**Comment by dkwl93** [comment]
Hey all, I found this issue while googling.. Has anyone managed to successfully build Android apps on AppCenter (React native version 0.61.5)?

**Comment by geekazoid-at** [comment]
> Hey all, I found this issue while googling.. Has anyone managed to successfully build Android apps on AppCenter (React native version 0.61.5)?

Yes, I managed to do it somehow, but actually I can't recall right now which bandaid made it build. I was in contact with AppCenter support, but they were not interested in how I actually fixed it. It was one of the dozen possible fixes that can be found on github. (that made me especially sad).

If somebody else remembers please help @dkwl93. If nobody answers, feel free to contact me again and I'll dig it out of SCM, but I am currently very very busy so I won't right away.

But in general, Android RN 0.65.1 builds with the correct bandaid.

**Comment by ddolheguy** [comment]
@geekazoid-at @dkwl93 I too would be interested and AppCenter Support although did contact me, were not interested in helping and just advised to wait for the next RN update :/

**Comment by geekazoid-at** [comment]
> @geekazoid-at @dkwl93 I too would be interested and AppCenter Support although did contact me, were not interested in helping and just advised to wait for the next RN update :/

Here you go, let me know if it helped. This seems to be the commit that fixed it for me:

Added this line to settings.gradle.
```
 rootProject.name = 'traceforce'
+
+apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
+
 include ':react-native-localize'
 project(':react-native-localize').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-localize/android')
 include ':@react-native-community_async-storage'
```

Fixed cli-platform-android version in package json.
```
     "@babel/core": "^7.6.2",
     "@babel/preset-env": "^7.0.0",
     "@babel/register": "^7.0.0",
+    "@react-native-community/cli-platform-android": "3.0.2",
     "@react-native-community/eslint-config": "^0.0.5",
     "babel-eslint": "9.0.0",
     "babel-jest": "^24.9.0",
```

This resulted in a working build and valid/non-crashing app package. Some of my other tries resulted in a working build but crashing app on startup.

**Comment by dkwl93** [comment]
Thanks so much @geekazoid-at @ddolheguy for the quick response! I'll try this tonight and report back here with results :) Hopefully other people with the same problem stumble upon this thread

**Comment by ddolheguy** [comment]
@dkwl93 Just adding my 2c here.

I've found specifically that version 3.0.2 of the "@react-native-community/cli-platform-android" is the only one which works for me.  Upgrading even to 3.0.3 or later falls back to the original issue.

**Comment by dkwl93** [comment]
Hey all! I've managed to fix builds by using a combination of @geekazoid-at solution and turning on universal builds `def enableSeparateBuildPerCPUArchitecture = false` in `app/build.gradle`


Thanks to all that contributed <3

**Comment by jkoutavas** [comment]
Thanks @geekazoid-at
